addons/api/addon/serializers/application.js

```diff
@@ -74,18 +75,33 @@ export default class ApplicationSerializer extends RESTSerializer {
       delete json[key];
     }
     // If an attribute has a `for` option, it must match the
-    // record's `type`, else the attribute is excluded
+    // record's `type`, otherwise the attribute is excluded
     // from serialization.
-    //note: This doesn't handle secrets yet, we can add support for them if needed.
-    if (
-      options?.for &&
-      options.for !== recordType &&
-      !options.for.includes(recordType)
-    ) {
-      if (options.isNestedAttribute) {
-        delete json?.attributes?.[key];
-      } else {
-        delete json[key];
+
+    if (options?.for) {
+      // Check the typeof the `for` option to determine if it is a plugin or not
+      const isPlugin =
+        typeOf(options.for) === 'object' && options.for?.type === 'plugin';
+      const isNotPlugin = typeOf(options.for) === 'string';
+
+      // Helper function to handle deleting attribute and secret keys
+      const deleteKey = (json) => {
+        if (options.isNestedAttribute) {
+          delete json?.attributes?.[key];
+        } else if (options.isNestedSecret && json?.secrets?.[key]) {
+          delete json?.secrets?.[key];
```

Comment:
Do we have to handle the case where it's only one key and deleting it would leave an empty object? Also same question in the host-catalog serializer

Reply:
right now I didn't handle empty cases as I noticed we do send `attributes: {}` to the api [here](https://github.com/hashicorp/boundary-ui/blob/ca4aad0a2b381257e6910e22f11f626da8e4c2eb/addons/api/tests/unit/serializers/managed-group-test.js#L35). Also checking this with foundations team if we need to clean this up

Reply:
empty `attributes: {}` is expected, but we need to clean up empty secrets. I pushed up the change for this one.